hibernate-core/src/test/java/org/hibernate/orm/test/mapping/hhh17404/OracleOsonCompatibilityTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.mapping.hhh17404;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.cfg.DialectSpecificSettings;
+import org.hibernate.dialect.OracleDialect;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.RequiresDialect;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.hibernate.type.SqlTypes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * This test class is about testing that legacy schema that use BLOB or CLOB for JSON columns
+ * can be safely read even when Oracle Oson extension is in place.
+ * In Such a situation, the JSON type will expect JSON a JSON column and should
+ * silently fall back to String deserialization.
+ *
+ * @author Emmanuel Jannetti
+ */
+@DomainModel(annotatedClasses = OracleOsonCompatibilityTest.JsonEntity.class)
+@SessionFactory(exportSchema = false)
+@RequiresDialect( value = OracleDialect.class, majorVersion = 23 )
+public abstract class OracleOsonCompatibilityTest {
+
+	@ServiceRegistry(settings = @Setting(name = DialectSpecificSettings.ORACLE_OSON_DISABLED, value = "true"))
+	public static class OracleOsonAsUtf8CompatibilityTest extends OracleOsonCompatibilityTest {
+		public OracleOsonAsUtf8CompatibilityTest() {
+			super( "JSON" );
+		}
+	}
+
+	public static class OracleBlobAsOsonCompatibilityTest extends OracleOsonCompatibilityTest {
+		public OracleBlobAsOsonCompatibilityTest() {
+			super( "BLOB" );
+		}
+	}
+
+	public static class OracleClobAsOsonCompatibilityTest extends OracleOsonCompatibilityTest {
+		public OracleClobAsOsonCompatibilityTest() {
+			super( "CLOB" );
+		}
+	}
+
+
+	private final String jsonType;
+
+	private final LocalDateTime theLocalDateTime = LocalDateTime.of( 2000, 1, 1, 0, 0, 0 );
+	private final LocalDate theLocalDate = LocalDate.of( 2000, 1, 1 );
+	private final LocalTime theLocalTime = LocalTime.of( 1, 0, 0 );
+	private final UUID uuid = UUID.fromString("53886a8a-7082-4879-b430-25cb94415be8");
+	private final String theString = "john";
+
+	public OracleOsonCompatibilityTest(String jsonType) {
+		this.jsonType = jsonType;
+	}
+
+	@BeforeEach
```

Comment:
Please add back the tear down method, otherwise that table will stick around after a test:

```java
	@AfterEach
	public void tearDown(SessionFactoryScope scope) {
		scope.inTransaction(
				(session) -> {
					session.createNativeQuery( getDialect().getDropTableString( "TEST_OSON_COMPAT" ) ).executeUpdate();
				}
		);
	}
```